lib/Graphs/ICFG.cpp

```diff
@@ -460,11 +472,14 @@ struct DOTGraphTraits<ICFG*> : public DOTGraphTraits<PAG*>
         if (IntraBlockNode* bNode = SVFUtil::dyn_cast<IntraBlockNode>(node))
         {
             rawstr << "IntraBlockNode ID: " << bNode->getId() << " \t";
-            PAG::PAGEdgeList&  edges = PAG::getPAG()->getInstPTAPAGEdgeList(bNode);
-            for (PAG::PAGEdgeList::iterator it = edges.begin(), eit = edges.end(); it != eit; ++it)
-            {
-                const PAGEdge* edge = *it;
-                rawstr << edge->toString();
+            rawstr << value2String(bNode->getInst()) << " \t";
+            if (Options::IncludePAGInICFGDump) {
+                PAG::PAGEdgeList&  edges = PAG::getPAG()->getInstPTAPAGEdgeList(bNode);
+                for (PAG::PAGEdgeList::iterator it = edges.begin(), eit = edges.end(); it != eit; ++it)
+                {
+                    const PAGEdge* edge = *it;
+                    rawstr << edge->toString();
```

Comment:
One more thing, unfortunately,  `toString()` will also output the LLVM instruction, which makes the node label with duplicate instruction. See here: https://github.com/SVF-tools/SVF/blob/39385bbb18b4b11ebdd4be5b4290064f9200dca0/lib/Graphs/PAG.cpp#L53-L67

I would suggest make a `if` and `else` branch:
```
if (Options::IncludePAGInICFGDump) {
...
rawstr << edge->toString();
}
else{
rawstr << value2String(bNode->getInst()) << " \t";
}
```

Any better idea?

Reply:
What you have was needed, but also if there are no PAG edges the instruction must be added explicitly.